GeneralsMD/Code/GameEngineDevice/Source/W3DDevice/GameClient/Shadow/W3DProjectedShadow.cpp

```diff
@@ -1923,20 +1919,19 @@ W3DProjectedShadow* W3DProjectedShadowManager::createDecalShadow(Shadow::ShadowT
 	Real	decalOffsetX=0.0f;
 	Real	decalOffsetY=0.0f;
 	const Real defaultWidth = 10.0f;
-
-	Char	texture_name[64];
-	Int nameLen;
+	Char texture_name[ARRAY_SIZE(shadowInfo->m_ShadowName)];
 
 	//simple decal using the premade texture specified.
 	//can be always perpendicular to model's z-axis or projected
 	//onto world geometry.
-	nameLen=strlen(shadowInfo->m_ShadowName);
-	if (nameLen <= 1)	//no texture name given, use same as object
-	{	strcpy(texture_name,defaultDecalName);
+	if (strlen(shadowInfo->m_ShadowName) <= 1)	//no texture name given, use same as object
```

Comment:
I expect we can simplify this to `if (shadowInfo->m_ShadowName[0] == '\0')` without any change in functionality.